libs/core/config/include/hpx/config.hpp

```diff
@@ -208,7 +208,7 @@
 #endif
 #if !defined(HPX_HAVE_SPINLOCK_DEADLOCK_DETECTION)
 #  if defined(HPX_DEBUG)
-//#    define HPX_HAVE_SPINLOCK_DEADLOCK_DETECTION
+#    define HPX_HAVE_SPINLOCK_DEADLOCK_DETECTION
```

Comment:
I think this has been disabled, unless explicitly asked for in the CMake configuration, because it may lead to false positives (if spinning just takes a very long time). I'm not sure it's a problem in 99.9% of the cases, but I just want to mention this since this will always enable it for debug builds with no way to turn it off.

Reply:
I have removed the silent configuration change in Debug mode now and have moved this to explicit options in the build system. I think this is the correct solution...

Reply:
Thanks! The downside of this is that the default will persist when changing the build type which is a bit unfortunate. I don't think there's a right solution here, but I think your current change is ok.